spec/firebase_app_distribution_api_client_spec.rb

```diff
@@ -441,4 +441,95 @@
         .to raise_error("#{ErrorMessage::INVALID_TESTERS} \nEmails: #{emails} \nGroups: #{group_ids}")
     end
   end
+
+  describe '#add_testers' do
+    let(:headers) { { 'Authorization' => 'Bearer auth_token', 'Content-Type' => 'application/json' } }
+
+    it 'is successful' do
+      emails = %w[1@foo.com 2@foo.com]
+      stubs.post("/v1/projects/project_number/testers:batchAdd", { emails: emails }.to_json, headers) do |env|
+        [
+          200,
+          {},
+          {}
+        ]
+      end
+
+      result = api_client.add_testers("project_number", emails)
+
+      expect(result.success?).to eq(true)
+    end
+
+    it 'fails and prints correct error message for a 400' do
+      emails = %w[foo 2@foo.com]
+      stubs.post("/v1/projects/project_number/testers:batchAdd", { emails: emails }.to_json, headers) do |env|
+        [
+          400,
+          {},
+          {}
+        ]
+      end
+
+      expect { api_client.add_testers("project_number", emails) }
+        .to raise_error(ErrorMessage::INVALID_EMAIL_ADDRESS)
+    end
+
+    it 'fails and prints correct error message for a 404' do
+      emails = %w[1@foo.com 2@foo.com]
+      stubs.post("/v1/projects/bad_project_number/testers:batchAdd", { emails: emails }.to_json, headers) do |env|
+        [
+          404,
+          {},
+          {}
+        ]
+      end
+      expect { api_client.add_testers("bad_project_number", emails) }
+        .to raise_error(ErrorMessage::INVALID_PROJECT)
+    end
+
+    it 'fails and prints correct error message for a 429' do
+      emails = %w[1@foo.com 2@foo.com]
+      stubs.post("/v1/projects/project_number/testers:batchAdd", { emails: emails }.to_json, headers) do |env|
+        [
+          429,
+          {},
+          {}
+        ]
+      end
+      expect { api_client.add_testers("project_number", emails) }
+        .to raise_error(ErrorMessage::TESTER_LIMIT_VIOLATION)
+    end
+  end
+
+  describe '#remove_testers' do
+    let(:headers) { { 'Authorization' => 'Bearer auth_token', 'Content-Type' => 'application/json' } }
+
+    it 'is successful' do
```

Comment:
How about `'returns the number of testers successfully removed'`

Reply:
Done.